cmd/plume/prerelease.go

```diff
@@ -0,0 +1,238 @@
+
+	plog.Printf("Pre-release complete, run `plume release` to finish.")
+
+	return nil
+}
+
+// getAzureVhd downloads a CoreOS image for Azure and unzips it to vhdPath.
+func getAzureVhd(spec *channelSpec, client *http.Client, src *storage.Bucket, bzipPath, vhdPath string) error {
+	if _, err := os.Stat(vhdPath); err == nil {
+		plog.Printf("Reusing existing image %q", vhdPath)
+		return nil
+	}
+
+	vhduri, err := url.Parse(spec.Azure.Image)
+	if err != nil {
+		return err
+	}
+
+	vhduri = src.URL().ResolveReference(vhduri)
+
+	plog.Printf("Downloading Azure image %q to %q", vhduri, bzipPath)
+
+	if err := sdk.UpdateSignedFile(bzipPath, vhduri.String(), client); err != nil {
+		return err
+	}
+
+	// decompress it
+	plog.Printf("Decompressing %q...", bzipPath)
+	return util.Bunzip2File(vhdPath, bzipPath)
+}
+
+func createAzureImage(spec *channelSpec, api *azure.API, blobName, imageName string) error {
+	imageexists, err := api.OSImageExists(imageName)
+	if err != nil {
+		return fmt.Errorf("failed to check if image %q exists: %v", imageName, err)
+	}
+
+	if imageexists {
+		plog.Printf("OS Image %q exists, using it", imageName)
+		return nil
+	}
+
+	plog.Printf("Creating OS image with name %q", imageName)
+
+	bloburl := fmt.Sprintf("https://%s.blob.core.windows.net/%s/%s", spec.Azure.StorageAccount, spec.Azure.Containers[0], blobName)
+
+	// a la https://github.com/coreos/scripts/blob/998c7e093922298637e7c7e82e25cee7d336144d/oem/azure/set-image-metadata.sh
+	md := &azure.OSImage{
+		Label:             spec.Azure.Label,
+		Name:              imageName,
+		OS:                "Linux",
+		Description:       spec.Azure.Description,
+		MediaLink:         bloburl,
+		ImageFamily:       spec.Azure.Label,
+		PublishedDate:     time.Now().UTC().Format("2006-01-02"),
+		RecommendedVMSize: spec.Azure.RecommendedVMSize,
+		IconURI:           spec.Azure.IconURI,
+		SmallIconURI:      spec.Azure.SmallIconURI,
+	}
+
+	return api.AddOSImage(md)
+}
+
+func replicateAzureImage(api *azure.API, imageName string) error {
+	plog.Printf("Fetching Azure Locations...")
+	locations, err := api.Locations()
+	if err != nil {
+		return err
+	}
+
+	plog.Printf("Replicating image to locations: %s", strings.Join(locations, ", "))
+
+	if err := api.ReplicateImage(imageName, "CoreOS", specChannel, specVersion, locations...); err != nil {
+		return fmt.Errorf("image replication failed: %v", err)
+	}
+
+	return nil
+}
+
+// azurePreRelease runs everything necessary to prepare a CoreOS release for Azure.
+//
+// This includes uploading the vhd image to Azure storage, creating an OS image from it,
+// and replicating that OS image.
+func azurePreRelease(ctx context.Context, client *http.Client, src *storage.Bucket, spec *channelSpec) error {
+	if spec.Azure.StorageAccount == "" {
+		plog.Notice("Azure image creation disabled.")
+		return nil
+	}
+
+	// construct azure api client
+	plog.Printf("Creating Azure API from %q", azureOpts.PublishSettingsFile)
+	a, err := azure.New(&azureOpts)
```

Comment:
Can you use a more descriptive name? I realize it's not idiomatic, but that idiom is idiotic.